src/main.rs

```diff
@@ -1,22 +1,135 @@
 use conways::Conways;
+use macroquad::prelude::*;
 
 mod conways;
 
-fn main() {
-    let mut conway = Conways::new(6, 6);
-    conway.set_alive(1,0);
-    conway.set_alive(1,1);
-    conway.set_alive(1,2);
+const CELL_SIZE: f32 = 10.0;
+const UPDATE_INTERVAL: f64 = 1.0;
+const WIDTH: usize = 50;
+const HEIGHT: usize = 50;
 
-    conway.set_alive(5,5);
-    conway.set_alive(5,4);
-    conway.set_alive(4,4);
-    conway.set_alive(4,5);
+struct Game {
+    conways: Conways,
+    previous_time: f64,
+    width: usize,
+    height: usize,
+    state: GameState
+}
+
+#[derive(PartialEq)]
+enum GameState {
+    Running,
+    Paused
+}
+
+impl Game {
+    pub fn new(width: usize, height: usize) -> Game {
+        Game {
+            conways: Conways::new(width, height),
+            previous_time: get_time(),
+            width,
+            height,
+            state: GameState::Running
+        }
+    }
+
+    pub fn update(&mut self) {
+        let time_of_last_frame = get_time();
+        self.draw();
+
+        // rendering time is too fast to be visually pleasing
+        // update cells state in fixed preriods of time
+        if 
+            time_of_last_frame - self.previous_time > UPDATE_INTERVAL
+            && self.state == GameState::Running 
+        {
+            self.conways.tick();
+            self.previous_time = time_of_last_frame;
+        }
+    }
+
+    fn draw(&self) {
+        clear_background(BLACK);
+        for x in  0..self.height {
+            for y in 0..self.width {
+                if self.conways.is_alive(x,y) {
+                    draw_rectangle(
+                        x as f32 * CELL_SIZE, 
+                        y as f32 * CELL_SIZE, 
+                        CELL_SIZE, 
+                        CELL_SIZE, 
+                        WHITE);
+                }
+            }
+        }
+    }
+
+    // handles all interaction with the user
+    pub fn user_interaction(&mut self) {
+        if is_key_pressed(KeyCode::Space) {
+            self.state = match self.state {
+                GameState::Paused => GameState::Running,
+                GameState::Running => GameState::Paused
+            };
+            
+        }
+
+        // Only can add or remove cells when paused
+        // else it wouldn't be easy to form patterns
+        if self.state == GameState::Paused {
+            if is_mouse_button_pressed(MouseButton::Left) {
+                self.set_alive();
+            } else if is_mouse_button_pressed(MouseButton::Right) {
+                self.set_dead();
+            }
+        }
+    }
+
+    fn set_dead(&mut self) {
+        let (x, y) = get_mouse_grid_position();
+        self.conways.set_dead(x, y);
+    }
+    
+    fn set_alive(&mut self) {
+        let (x, y) = get_mouse_grid_position();
+        self.conways.set_alive(x, y);
+    }
```

Comment:
Same as above, you could use a set_state function, and one argument is the state